packages/app-desktop/gui/Navigator.jsx

```diff
@@ -47,8 +55,29 @@ class NavigatorComponent extends Component {
 }
 
 const Navigator = connect(state => {
+	let currentNote = null;
+	let currentFolder = null;
+	if (state.selectedNoteIds.length) {
+		const noteId = state.selectedNoteIds[0];
+		const note = state.notes.find((_note) => _note.id == noteId);
+		if (note) {
+			currentNote = {
+				id: noteId,
+				title: note.title,
+			};
+		}
+	}
+	if (state.selectedFolderId) {
+		const folder = state.folders.find((_folder) => _folder.id == state.selectedFolderId);
+		if (folder) {
+			currentFolder = {
+				id: state.selectedFolderId,
+				title: folder.title,
+			};
+		}
+	}
```

Comment:
We don't add derived state to the state because you can get all this at the point where you need it. In fact, please create a "windowTitle()" function that encapsulates that new logic and the existing logic and returns the window title. Having a separate function will also make it easier to unit test it.